src/base/BaseTool.js

```diff
@@ -94,6 +101,26 @@ export default class BaseTool {
     }
   }
 
+
+  /**
+   * Applys the requested mixins to the class.
+   *
+   * @param  {Array} mixinArray An array of mixin names (strings).
+   */
+  _applyMixins (mixins) {
+    for (let i = 0; i < mixins.length; i++) {
+      const mixinName = mixins[i];
+
+      if (typeof mixinCollection[mixinName] === 'object') {
+        Object.assign(this, mixinCollection[mixinName]);
```

Comment:
Only because I'm way overthinking it, but what if the user has two mixins that want to add a behavior that's called in the `passiveCallback`? The last one applied would override. For each callback, do we want to allow an _array of functions_, where each one is called after the other?

I mean.. Probably, maybe, eventually.
Probably not worth adding in just yet as we don't have any use cases for it yet.
Just food for thought ^_^